docs/DeveloperGuide.md

```diff
@@ -342,6 +338,52 @@ Step 6. The user executes a `remove s/3` command which removes the instance of `
+
+The attendance feature is facilitated by `AttendanceCommand`, `AttendanceList` and `AttendanceList` classes.
+`AttendanceCommand` extends `Command` abstract class that has  abstract method, `execute()`.
+`Attendance` deals with the actual attendance inputs while the
+`AttendanceList` deals with the list of `Attendance`.
+The `AttendanceList` is the object that is displayed in the `AddressBook`.
+Additionally, the classes implement the following operations:
+- `AttendanceList#addAttendance(Attendance)` - Adds attendance to the attendance list.
+- `AttendanceList#clearList()` - Clears all stored attendance of the Person
+- `AttendanceList#editAtIndex(index)` - Edits the attendance at the given index with the new given attendance.
+
+These operations are exposed in command executions such as `AttendanceCommand#execute()` and `EditCommand#createEditedPerson()`.
+`Attendance#clearList()` is used for testing purposes only.
+
+Given below is an example usage scenario and how the adding attendance mechanism behaves at each step.
+
+Step 1. The user launches the application for the first time. The address book will be initialized with the initial address book state.
+
+Step 2. The user executes `attendance 1 a/2022-08-08` to add attendance to the first `Person`'s attendance list in the address book.
+The `Attendance` command calls `AttendanceList#addAttendance(Attendance)` and adds the task to the list.
+
+The following sequence shows how adding attendance works:
+
+![AttendanceSequenceDiagram](images/AttendanceSequenceDiagram.png)
+
+<div markdown="span" class="alert alert-info">
+:information_source: **Note** The lifeline for `AttendanceCommandParser` should end at the destroy marker (X) but due to a limitation of PlantUML, the lifeline reaches the end of diagram.
+</div>
+
+Step 3. The user decides to edit the attendance to 2022-08-12 instead.
+They go into full view mode with the `view` command and execute `edit 1 a/2022-08-12`, which calls `AttendanceList#editAtIndex()` and replaces the old description with the new one.
+
+#### Design considerations:
+
+**Aspect: Format of user input in Attendance**
+
+* **Alternative 1 (current choice):** Use `LocalDateTimeFormatter` as formatter for `LocalDate`.
+    * Pros: Easier to implement, uses only one imported Java class `LocalDate` for encapsulating user input and intuitive for user.
+    * Cons: Cannot differentiate between `Session` timings on the same day. Hence, a `Person` who has two `Session` on the same day, who is present for one or the other cannot be differentiated. 
+
+* **Alternative 2:** Use of `LocalDateTime` instead of `LocalDate`.
+    * Pros: Able to differentiate `Attendance` between two `Session` timings on the same day.
+    * Cons: Longer user input. This results in longer user input every time the user adds an `Attendance` to a `Person`. May result in cumulative inconvenience over a long period of time. Can be solved by initial implementation by adding two `Attendance`.
```

Comment:
Title of your Alternative 1 is weird. You are talking about using `LocalDate` instead of `LocalDateTime` (with the opposite being Alternative 2) so not sure why you are bringing up `LocalDateTimeFormatter`

Reply:
thank you have changed it